File: javascript/06-nested-if-hell/src/loan_approver.js
class LoanApprover {
  constructor() {}

  can_deliver(pkg) {
    if (pkg) {
      if (pkg.weight) {
        if (pkg.weight <= 50) {
          if (pkg.hazardous === false) {
            if (pkg.temperatureRequired) {
              if (pkg.temperatureRequired >= -20 && pkg.temperatureRequired <= 40) {
                if (pkg.weekend === false) {
                  if (pkg.remoteArea) {
                    if (pkg.remoteArea === true) {
                      if (pkg.weight <= 20) {
                        return { allowed: true, warning: 'Remote surcharge applies' };
                      } else {
                        return { allowed: false, warning: 'Too heavy for remote' };
                      }
                    } else {
                      return { allowed: true, warning: null };
                    }
                  } else {
                    return { allowed: true, warning: null };
                  }
                } else {
                  return { allowed: false, warning: 'No weekend delivery' };
                }
              } else {
                return { allowed: false, warning: 'Temperature out of range' };
              }
            } else {
              if (pkg.weekend === false) {
                if (pkg.remoteArea) {
                  if (pkg.remoteArea === true) {
                    if (pkg.weight <= 20) {
                      return { allowed: true, warning: 'Remote surcharge applies' };
                    } else {
                      return { allowed: false, warning: 'Too heavy for remote' };
                    }
                  } else {
                    return { allowed: true, warning: null };
                  }
                } else {
                  return { allowed: true, warning: null };
                }
              } else {
                return { allowed: false, warning: 'No weekend delivery' };
              }
            }
          } else {
            return { allowed: false, warning: 'Hazardous material' };
          }
        } else {
          return { allowed: false, warning: 'Weight exceeded' };
        }
      } else {
        return { allowed: false, warning: 'No weight specified' };
      }
    } else {
      return { allowed: false, warning: 'No package' };
    }
  }
}

export { LoanApprover };

File: python/06-nested-if-hell/src/loan_approver.py
class LoanApprover:
    def __init__(self):
        pass

    def can_deliver(self, pkg):
        if pkg:
            if pkg.get("weight"):
                if pkg["weight"] <= 50:
                    if pkg.get("hazardous") is False:
                        if pkg.get("temperatureRequired"):
                            if pkg["temperatureRequired"] >= -20 and pkg["temperatureRequired"] <= 40:
                                if pkg.get("weekend") is False:
                                    if pkg.get("remoteArea"):
                                        if pkg["weight"] <= 20: return {"allowed": True, "warning": "Remote surcharge applies"}
                                        else: return {"allowed": False, "warning": "Too heavy for remote"}
                                    return {"allowed": True, "warning": None}
                                return {"allowed": False, "warning": "No weekend delivery"}
                            return {"allowed": False, "warning": "Temperature out of range"}
                        else:
                            if pkg.get("weekend") is False:
                                if pkg.get("remoteArea"):
                                    if pkg["weight"] <= 20: return {"allowed": True, "warning": "Remote surcharge applies"}
                                    else: return {"allowed": False, "warning": "Too heavy for remote"}
                                return {"allowed": True, "warning": None}
                            return {"allowed": False, "warning": "No weekend delivery"}
                    return {"allowed": False, "warning": "Hazardous material"}
                return {"allowed": False, "warning": "Weight exceeded"}
            return {"allowed": False, "warning": "No weight specified"}
        return {"allowed": False, "warning": "No package"}
